### PR TITLE
BREAKING: FluentProvider now sets a default css `line-height`

### PR DESCRIPTION
**Pull request by behowell**
## Current Behavior

FluentProvider sets the font family, size, and weight, but does not set line height. Plain body text inside a FluentProvider context doesn't match the design spec.

## New Behavior

FluentProvider now sets `line-height`.

This has the potential to change layout of any text in an app that doesn't explicitly set `line-height`. In some cases, this will result in fixing layout bugs, but it could also cause issues if a component intended to change the typography but didn't also specify `line-height`.

## Related Issue(s)

This splits out just the FluentProvider change from @andrefcdias PR #23254. In discussion with the team, we decided to hold on the `Text` component changes from the PR pending further discussion. However, we don't want to delay the `FluentProvider` change before the `Text` decision is finalized.

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/Avatar-mount_base.html">916</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/Avatar-mount.html">901</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/Button-mount_base.html">564</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/Button-mount.html">556</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/FluentProvider-mount_base.html">1903</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/FluentProvider-mount.html">1840</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/FluentProviderWithTheme-mount_base.html">278</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/FluentProviderWithTheme-mount.html">276</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">242</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">238</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">308</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">324</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/MakeStyles-mount_base.html">1588</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23279/perf-test-react-components/MakeStyles-mount.html">1593</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by behowell** [comment]
Regarding the screener regressions, I think it should be ok to approve of all of them:
* **Badge**/**CounterBadge**
  * These component's styles need to be fixed in general, as they don't match the design specs before or after this PR.
  * This will be done in a follow-up PR; tracked by #23278
* **Card**
  * The changes to the body text inside the card are actually being _fixed_ by this PR. The line height is now correct.
  * One of the screener stories used `<h1>` elements internally, which ended up with the wrong line height after this change. However, after looking through the tests, those scenarios were entirely redundant with other stories, so I removed them.
* **Image**
  * The broken image alt text now has the same line height as body text. If anything, it's more correct now (and really doesn't matter).
* **MakeStyles** and **Positioning**
  *  These tests are using some filler text, which now has the right line height. It doesn't affect production code.

File: apps/vr-tests-react-components/src/stories/Card.stories.tsx
Comment (behowell) [inline apps/vr-tests-react-components/src/stories/Card.stories.tsx:93]: These stories use the `<h1>` tag, which was ending up with the wrong line height after this change (bug in the test). It could have been replaced by `<Title1 as="h1">`; however, this is all entirely redundant with the "appearance interactive - Filled", etc. stories below, which are testing the same appearances. So I deleted them.

Comment (andrefcdias) [reply apps/vr-tests-react-components/src/stories/Card.stories.tsx:93]: This is NOT redundant. Filled and Interactive Filled use different styles, see https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-card/src/components/Card/useCardStyles.ts#L54.
Please add the test back and either change the headers or let the visual regression test fail, which means it's doing it's job correctly, and we can approve as it's intended.

**Review by andrefcdias — CHANGES_REQUESTED** [review]
Theme provider changes are good, but the Card stories need to be re-added

**Review by jurokapsiar — APPROVED** [review]

File: apps/vr-tests-react-components/src/stories/Card.stories.tsx
Comment (behowell) [reply apps/vr-tests-react-components/src/stories/Card.stories.tsx:93]: Ok I'll revert these changes, thanks!

**Comment by behowell** [comment]
@andrefcdias I reverted the Card changes. I also accepted all of the screener changes and set the PR to auto-merge after you approve, if it looks good. Thanks!

**Pull request merged**
